Client/Frontend/Browser/BrowserViewController/BrowserViewController+InterceptorUI.swift

```diff
@@ -32,4 +32,11 @@ extension BrowserViewController: InterceptorUI {
 
         self.present(alert, animated: true, completion: nil)
     }
+
+    func openInPrivateMode(tab: Tab, url: URL, policy: InterceptorPolicy) {
```

Comment:
This method does not belong here. I would like to use this opportunity to establish a `UseCase` architectural block. Following the example of mozilla/reference browser I would like to have a useCase object attached to Application Delegate, so any feature can easily access it. https://github.com/mozilla-mobile/reference-browser/blob/master/app/src/main/java/org/mozilla/reference/browser/Components.kt#L22

Reply:
The 'new private tab' in mozilla codebase is implemented here https://github.com/mozilla-mobile/android-components/blob/master/components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt#L116